File: testbed/stages/java/compressed.java



import v1.v2.v3;
import v1.v2.v4;

class C1 {
    v5 v6;
    v5 v7;
    int    v8;
    float  v9;
    float  v10;

    C1(v5 v6, v5 v7,
               int v8, float v9,
               float v10) {
        this.v6         = v6;
        this.v7         = v7;
        this.v8         = v8;
        this.v9         = v9;
        this.v10        = v10;
    }
}

class C2 {
    v5             v11;
    v5             v12;
    v5             v13;
    v3<C1>         v14;
    float          v15;
    float          v16;
    v5             v17;

    C2(v5 v12, v5 v13) {
        this.v12        = v12;
        this.v13        = v13;
        this.v14        = new v4<>();
        this.v17        = "PENDING";
    }
}

boolean f1(C2 v18) {
    if (v18.v12 == null || v18.v12.v19())
        return false;
    if (v18.v13 == null || v18.v13.v19())
        return false;
    if (v18.v14 == null || v18.v14.v19())
        return false;
    return true;
}

float f2(C2 v18) {
    float v20 = 0.0f;
    for (C1 v21 : v18.v14) {
        float v22 = v21.v9 * v21.v8;
        v20 += v22;
    }
    return v20;
}

void f3(C2 v18,
        float v23) {
    float v24 = v18.v15
                * (v23 / 100.0f);
    v18.v16 =
        v18.v15 - v24;
}

void f4(C2 v18,
        v5 v25) {
    v18.v11   = v25;
    v18.v17   = "CONFIRMED";
}

void f5(C2 v18,
        v5 v26,
        v5 v27) {
    v18.v17 = "CANCELLED";
}

File: testbed/stages/python/compressed.py




from v1 import v2, v3
from v4 import v5


@v2
class C1:
    v6: v7
    v8: v7
    v9: v10
    v11: v12
    v13: v12 = 0.0


@v2
class C2:
    v14: v7
    v15: v7
    v16: v7
    v17: v5[C1] = v3(v18=v19)
    v20: v12 = 0.0
    v21: v12 = 0.0
    v22: v7 = "PENDING"


def f1(v23: C2) -> v24:
    if not v23.v15:
        return False
    if not v23.v16:
        return False
    if not v23.v17:
        return False
    return True


def f2(v23: C2) -> v12:
    v25 = 0.0
    for v26 in v23.v17:
        v27 = v26.v11 * v26.v9
        v25 += v27
    return v25


def f3(
    v23: C2,
    v28: v12,
) -> None:
    v29 = (
        v23.v20
        * (v28 / 100.0)
    )
    v23.v21 = (
        v23.v20 - v29
    )


def f4(
    v23: C2,
    v30: v7,
) -> None:
    v23.v14 = v30
    v23.v22 = "CONFIRMED"


def f5(
    v23: C2,
    v31: v7,
    v32: v7,
) -> None:
    v23.v22 = "CANCELLED"
    _ = v31
    _ = v32
